firebase-functions/functions/src/choiceOperations.ts

```diff
@@ -0,0 +1,32 @@
+﻿import { Player } from './models/player';
+
+export function hasEveryoneChosen(players: Array<FirebaseFirestore.QueryDocumentSnapshot<FirebaseFirestore.DocumentData>>): boolean {
+  return players.every(player => (player.data() as Player).choice);
+}
+
+export function calculateLosers(players: Array<FirebaseFirestore.QueryDocumentSnapshot<FirebaseFirestore.DocumentData>>): string[] {
+  if (draw(players)) {
+    return [];
+  }
+  // this is just to show something. soon, all the losers will be calculated and returned here instead
+  return [players[0].id];
+}
+
+function draw(players: Array<FirebaseFirestore.QueryDocumentSnapshot<FirebaseFirestore.DocumentData>>): boolean {
+  return isEveryChoiceEqual(players) || isEveryChoiceDifferent(players);
+}
+
+export function getDistinctChoices(players: Array<FirebaseFirestore.QueryDocumentSnapshot<FirebaseFirestore.DocumentData>>): string[] {
+  return players
+    .map(playerDoc => (playerDoc.data() as Player).choice)
+    .filter((value, index, array) => array.indexOf(value) === index);
+}
+
+function isEveryChoiceEqual(players: Array<FirebaseFirestore.QueryDocumentSnapshot<FirebaseFirestore.DocumentData>>): boolean {
```

Comment:
`areAllTheChoicesEqual` maybe

Reply:
also here I think you can go with `getDistinctChoices(players).length === 1;` :)

Reply:
A very good suggestion, what was I thinking?